torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -396,7 +403,7 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
     InlineAutodiffSubgraphs(
         copy,
         getAutodiffSubgraphInlining() ? autodiffSubgraphInlineThreshold : 1);
-    RemoveProfilingNodes(copy);
+    replaceFallbackGraphWithFallbackFunction(copy->block());
```

Comment:
I was thinking that the fallback graph is inserted by insertTypeGuard using the subgraph attached to the guarded (ie the DifferentiableGraph) node. So it does not include the DifferentiableGraph itself. Then here the replaceFallbackGraphWithFallbackFunction replaces the fallback graph with the fallback function. (And that gets an executor so it can then do its own profiling + optimization.)